### PR TITLE
docs: add fluentd buffers vs DPM calculations info for metrics

### PR DESCRIPTION
**Pull request by perk-sumo**
##### Description

Fill in your description here.

---

##### Checklist

Remove items which don't apply to your PR.

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by swiatekm-sumo** [comment]
I think it'd be safer to divide the final values by 2 to account for uneven load on FluentD instances. Even with the remote write proxy, we're still relying on Kubernetes Service load balancing, which is far from perfect.

File: deploy/docs/Best_Practices.md [hunk 1]
```diff
@@ -255,6 +255,34 @@ See the following links to official Fluentd buffer documentation:
...
+minutes = (PV size in bytes * Fluentd instances) / (DPM * 333 bytes)
+```
+
+Example 1:  
+My cluster sends 10 thousand DPM to Sumo. I'm using default 10 gb of buffer size. I'm also using 3 Fluentd instances. That gives me 30 gb of buffers in total
+(3 * 10 gb). I'm using 3.33 mb per minute. My setup should be able to hold data for 9000 minutes, that is 150 hours or 6.25 days.
+
+Example 2:  
+My cluster sends 1 million DPM to Sumo. I'm using 20 gb of buffer size. I'm using 20 Fluentd instances. I have 400 gb of buffers in total (20 * 20 gb). I'm
+using 333 mb of buffer every minute. My setup should be able to hold data for around 1200 minutes, that is 20 hours.
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:284]: Strange line break

```suggestion
My cluster sends 1 million DPM to Sumo. I'm using 20 gb of buffer size. I'm using 20 Fluentd instances. I have 400 gb of buffers in total (20 * 20 gb).
I'm using 333 mb of buffer every minute. My setup should be able to hold data for around 1200 minutes, that is 20 hours.
```

File: deploy/docs/Best_Practices.md [hunk 2]
```diff
@@ -255,6 +255,34 @@ See the following links to official Fluentd buffer documentation:
...
+### Fluentd buffer size for metrics
+
+Should you have any connectivity problems, depending on the buffer size your setup will be able to survive for a given amount of time without a data loss,
+delivering the data later when everything is operational again.
+
+To calculate this time you need to know how much data you send. For the calculations below we made an assumption that a single metric data poing is around 1
+kilobyte in size, including metadata. This assumption is based on the average data we ingest. By default, for file based buffering we use gzip compression
+which gives us around 3:1 compress ratio.
+
+That results in `1 DPM` (Data Points per Minute) using around `333 bytes of buffer`. That is `333 kilobytes for 1 thousand DPM` and `333 megabytes for 1 million DPM`.
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:267]: Perhaps mention explicitly that this is storage used per minute.

File: deploy/docs/Best_Practices.md [hunk 3]
```diff
@@ -255,6 +255,34 @@ See the following links to official Fluentd buffer documentation:
 - https://docs.fluentd.org/configuration/buffer-section
 - https://docs.fluentd.org/buffer/file
 
+### Fluentd buffer size for metrics
+
+Should you have any connectivity problems, depending on the buffer size your setup will be able to survive for a given amount of time without a data loss,
+delivering the data later when everything is operational again.
+
+To calculate this time you need to know how much data you send. For the calculations below we made an assumption that a single metric data poing is around 1
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:263]:
```suggestion
To calculate this time you need to know how much data you send. For the calculations below we made an assumption that a single metric data point is around 1
```

Comment (perk-sumo) [reply deploy/docs/Best_Practices.md:263]: Fixed.

File: deploy/docs/Best_Practices.md (hunk 1 above)
Comment (perk-sumo) [reply deploy/docs/Best_Practices.md]: Fixed.

File: deploy/docs/Best_Practices.md (hunk 2 above)
Comment (perk-sumo) [reply deploy/docs/Best_Practices.md:267]: But that is already in minutes as DPM is in minutes 🤔 I'm not sure how to make that more clear.

**Comment by perk-sumo** [comment]
> This lgTM. Do we not want to have a section for logs/events as well?

We didn't check that for logs or events so cannot give such info yet.

**Pull request merged**